### PR TITLE
CSHARP-2691: Update current version in reference docs to 2.9

### PR DESCRIPTION
**Pull request by vincentkam**
Modeled after: https://github.com/rstam/mongo-csharp-driver/pull/112

### CODE REVIEW

File: Docs/reference/content/what_is_new.md [hunk 1]
```diff
@@ -13,6 +13,18 @@ title = "What's New"
 Some of the changes in 2.9.0 include:
 
 * Retryable writes are enabled by default
+* Retryable reads support (enabled by default)
```
Comment (vincentkam) [inline Docs/reference/content/what_is_new.md:16]: I tried to make this section more concise than our release notes, per previous style

**Review by jyemin — APPROVED** [review]
LGTM

**Comment by vincentkam** [comment]
Ready for review!

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Pull request merged**